olive/passes/onnx/gptq.py

```diff
@@ -0,0 +1,229 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import tempfile
+from typing import Any, Callable, Dict, List, Union
+
+import torch
+
+from olive.common.config_utils import validate_config
+from olive.data.config import DataConfig
+from olive.hardware.accelerator import AcceleratorSpec, Device
+from olive.model import PyTorchModelHandler
+from olive.model.handler.onnx import ONNXModelHandler
+from olive.model.utils.onnx_utils import resolve_onnx_path
+from olive.passes import Pass
+from olive.passes.onnx.common import model_proto_to_olive_model
+from olive.passes.onnx.conversion import OnnxConversion
+from olive.passes.pass_config import PassConfigParam
+
+
+class GptqQuantizer(Pass):
+    """GPTQ quantization using Hugging Face Optimum and export model with onnxruntime optimized kernel."""
+
+    _requires_user_script = True
+
+    @classmethod
+    def _default_config(cls, accelerator_spec: AcceleratorSpec) -> Dict[str, PassConfigParam]:
+        return {
+            "nsamples": PassConfigParam(
+                type_=int,
+                default_value=128,
+                description="number of samples in calibration dataset to apply quantization. Default value is 128",
+            ),
+            "bits": PassConfigParam(
+                type_=int,
+                default_value=4,
+                description="quantization bits. Default value is 4",
+            ),
+            "layers_block_name": PassConfigParam(
+                type_=str,
+                default_value="model.layers",
```

Comment:
Quick question: how user know which block they should use to quantize? It is a name stored in model? Or any fixed options user can choose?

Reply:
It's the name in model definition, and it's possible to have arbitrary name when custom model, so it maybe hard to provide options.  We also prefill these parameters for user for mainstream  models.
https://github.com/Taka152/Olive/blob/2361389e82bd8e468ce981603c26612c352bb267/olive/passes/onnx/gptq.py#L184-L197

Reply:
I added a link in argument description, it may help user to better set this args when using custom model.